x/gov/client/cli/tx.go

```diff
@@ -221,7 +242,17 @@ func GetCmdVote(cdc *wire.Codec) *cobra.Command {
 			}
 
 			msg := gov.NewMsgVote(voterAddr, proposalID, byteVoteOption)
-
+			if cliCtx.GenerateOnly {
```

Comment:
Can we encapsulate this logic in a common helper function?

Reply:
(also applies to other instances)

Reply:
ACKd and fixed.